homeassistant/components/tplink/binary_sensor.py

```diff
@@ -58,6 +58,10 @@ class TPLinkBinarySensorEntityDescription(
         key="water_alert",
         device_class=BinarySensorDeviceClass.MOISTURE,
     ),
+    TPLinkBinarySensorEntityDescription(
```

Comment:
I think this should be `event` platform.

Reply:
Potentially, yeah. But the event platform has its issues also, so I'd rather expose this as-is (if it works) and we can discuss about deprecating it later on.

I was just reading the community forums and noticed that many people were asking why there's no sensor for the motion detected, and there was a screenshot not showing the detected state. IIRC, one of the problems with events is that you won't get this either, but you must know that you are looking for a specific event.

Reply:
If the motion sensor has a duration, binary sensor is OK.

If the motion sensor is an momentary sensor than the event platform should be used.

Since we are polling for this, I think binary sensor is probably what we want